code/modules/mob/living/living_defense.dm

```diff
@@ -265,16 +265,30 @@
 
 /mob/living/proc/ExtinguishMob()
 	if(on_fire)
+		fire_stacks = min(0,fire_stacks)
 		on_fire = 0
-		fire_stacks = 0
 		set_light(max(0, light_range - MOB_FIRE_LIGHT_RANGE), max(0, light_power - MOB_FIRE_LIGHT_POWER))
 		update_fire()
 
 /mob/living/proc/update_fire()
 	return
 
-/mob/living/proc/adjust_fire_stacks(add_fire_stacks) //Adjusting the amount of fire_stacks we have on person
-    fire_stacks = Clamp(fire_stacks + add_fire_stacks, FIRE_MIN_STACKS, FIRE_MAX_STACKS)
+/mob/living/proc/adjust_fire_stacks(var/add_fire_stacks, var/should_ignite = FALSE, var/should_extinguish = FALSE, var/should_go_over=FALSE)
+
+	// add_fire_stacks: Amount of fire stacks to add. Could be positive or negative.
+	// should_ignite: Whether or not this change should ignire the person, if fire stacks are above 0.
+	// should_extinguish: Whether or not this change should extinguish the person, if fire stacks are below 0.
+	// should_go_over: Whether or not the change is allowed to go pass 0, the threshold between on fire and not on fire.
+	
+	if(!fire_stacks)
+		fire_stacks = 0
+
+	fire_stacks = Clamp(fire_stacks + add_fire_stacks, (!should_go_over && fire_stacks > 0) ? 0 : FIRE_MIN_STACKS, (!should_go_over && fire_stacks < 0) ? 0 : FIRE_MAX_STACKS)

```

Comment:
`should_go_over` sounds like a weird thing. A work around would be to just _not_ base fire off of `fire_stacks` but a simple boolean.

Reply:
I'm very confused when reading this suggestion. The current system is fine, and should_go_over effectively states why the fire stacks are being adjusted. Chemical changes uses should_go_over, while physical changes such as patting out fire doesn't.

Reply:
A function should do one thing, ideally. `adjust_x` series of procs are generally there to safely, in a clamped fashion, modify variables. They sometimes also ensure in call-stack and event integrity, but never in a fashion that's toggled based on their arguments.

This function is trying to do at least three things:
* adjust fire_stacks in a safe manner;
* decide and extinguish a mob, if need be;
* decide and light a mob on fire, if need be.

What's worse is that `should_extinguish` and `should_ignite` conflict with `should_go_over`, since the latter controls the preconditions that the former two require. It's a logical hellhole of an API. Which is why I would suggest leaving this function as either a dummy, and just making ignition sources call a complimentary `IgniteMob` (and extinguishers their equivalent) if need be, or in having a superior function for handling the three distinct logical components.